settings.gradle.kts

```diff
@@ -1,15 +1,15 @@
 plugins {
-    id("com.gradle.enterprise") version "3.17"
+    id("com.gradle.develocity") version "3.17"
     id("org.gradle.toolchains.foojay-resolver-convention") version "0.8.0"
 }
 
-val isCiBuild = System.getenv("CI") != null
-
-gradleEnterprise {
+develocity {
     buildScan {
-        termsOfServiceUrl = "https://gradle.com/terms-of-service"
-        termsOfServiceAgree = "yes"
-        publishAlwaysIf(isCiBuild)
+        termsOfUseUrl = "https://gradle.com/terms-of-service"
+        termsOfUseAgree = "yes"
+        // TODO: workaround for https://github.com/gradle/gradle/issues/22879.
+        val isCI = providers.environmentVariable("CI").isPresent
+        publishing.onlyIf { isCI }
```

Comment:
They have exactly the same example here:

https://docs.gradle.com/enterprise/gradle-plugin/current/#publishing_based_on_criteria

Could we stick with `getenv` as they show? I agree providers are the way, but if not working, then I would stick with the old way.

is it possible there may be a different syntax, something like:

```
publishing.onlyIf(providers.environmentVariable("CI").orElse(false))
```

Reply:
It can't be inline, this is why I commented TODO here.

Reply:
Inlining `!System.getenv("CI").isNullOrEmpty()` is working but `providers.environmentVariable("CI").isPresent`, I can use the former if you prefer.